iceoryx_posh/source/roudi/service_registry.cpp

```diff
@@ -145,13 +155,13 @@ uint32_t ServiceRegistry::findIndex(const capro::ServiceDescription& serviceDesc
     return NO_INDEX;
 }
 
-void ServiceRegistry::getAll(ServiceDescriptionVector_t& searchResult) const noexcept
+void ServiceRegistry::applyToAll(cxx::function_ref<void(const ServiceDescriptionEntry&)> function) const noexcept
 {
     for (auto& entry : m_serviceDescriptions)
     {
         if (entry)
         {
-            searchResult.emplace_back(*entry);
+            function(*entry);
```

Comment:
Here we have the same problem. What happens when `function` is not set?

Please add here as well a check right in the beginning of the function.
```
if ( !callable ) {
  return;
}
```